docs/hadoop.adoc

```diff
@@ -4,17 +4,72 @@
 JanusGraph-Hadoop works with TinkerPop's hadoop-gremlin package for
 general-purpose OLAP.
 
-Here's a three step example showing some basic integrated JanusGraph-TinkerPop functionality.
+The current tutorial covers the following topics for setting up Spark environment to work with Janusgraph using Hadoop for doing OLAP.
 
-1. Manually define schema and then load the Grateful Dead graph from a TinkerPop Kryo-serialized binary file 
-2. Run a VertexProgram to compute PageRanks, writing the derived graph to `output/~g`
-3. Read the derived graph vertices and their computed rank values
+1. Bulk loading into backend data store with local spark (Spark dist which comes with JanusGraph)
+2. Bulk loading into backend data store using Spark Cluster in Standalone mode. (The tutorial expects that there is a up and running Spark Cluster as provided host)
+3. Reads from backend data store into Hadoopgraph for OLAP against local spark
+4. Reads from Cassandra backend into Hadoopgraph for OLAP against Spark Cluster in standalone mode.
 
 [NOTE]
-The examples in this chapter are based on running Spark in local mode. Additional configuration 
-is required when using Spark in standalone mode or when running Spark on YARN or Mesos.
+The examples in this chapter are based on running Spark in local mode or standalone cluster mode. Additional configuration
+is required when using Spark on YARN or Mesos.
 
-=== Defining defining schema and loading data
+===== Pre-Requisite check:
+HADOOP_CONF_DIR needs to be set in CLASSPATH from where you will be running OLAP queries. Hence this adds dependency that Hadoop needs to be installed/configured from the box where gremlin console is started.
+
+For reference purpose, Gremlin-consolse is considered to be driver here, and hence HADOOP_CONF_DIR needs to be added to CLASSPATH of gremlin console.
+
+[source, shell]
+----
+cd janusgraph-0.2.1-SNAPSHOT-hadoop2/bin
+vi gremlin.sh
+
+CP="$CP":$(find -L $LIB -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
+# Add the jars in $BIN/../lib that start with "janusgraph"
+CP="$CP":$(find -L $LIB -name 'janusgraph*.jar' | sort | tr '\n' ':')
+# Add the remaining jars in $BIN/../lib.
+CP="$CP":$(find -L $LIB -name '*.jar' ! -name 'janusgraph*' ! -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
+# Add the jars in $BIN/../ext (at any subdirectory depth)
+CP="$CP":$(find -L $EXT -name '*.jar' | sort | tr '\n' ':')
+# Add Hadoop Configuration directory to ClassPath, for me it is situated at /root/hadoop/etc/hadoop
+CP="$CP":/root/hadoop/etc/hadoop
+----
+
+Now that Hadoop configuration is added to Gremlin console CLASSPATH, we then next want to configure the Hadoop configuration in local machine to point to an up and running HDFS cluster.
+
+Editing just the *core-site.xml* in Hadoop configuration directory works. The sample *core-site.xml* with Hadoop Master / NameNode is as follows:
```

Comment:
We probably need to tell how to write hdfs core-site.xml here. It is out of our scope.  TinkerPop has a good section on the  HADOOP_CONF_DIR and hdfs config.
http://tinkerpop.apache.org/docs/3.3.3/reference/#_installing_hadoop_gremlin
Can we borrow?

Reply:
I like your idea of borrowing from Tinkerpop docs. Is there a source for the same which I can reference for borrowing the section? But I didn't see any section which tells how to configure core-site.xml. Should I use their doc source for Hadoop Gremlin and then add my section for sample core-site.xml @jerryjch  ?

Reply:
We probably don't need the instruction for a sample core-site.xml.  It is out of our scope. We don't point to the config.